File: backend/internal/domain/dosen.go
package domain

import "time"

type Dosen struct {
	ID        uint64    `db:"id"         json:"id"`
	UserID    *uint64   `db:"user_id"    json:"user_id,omitempty"`
	NIDN      string    `db:"nidn"       json:"nidn"`
	Nama      string    `db:"nama"       json:"nama"`
	Email     string    `db:"email"      json:"email"`
	Prodi     string    `db:"prodi"      json:"prodi"`
	CreatedAt time.Time `db:"created_at" json:"created_at"`
	UpdatedAt time.Time `db:"updated_at" json:"updated_at"`
}

File: backend-refactor-python/app/schemas/dosen.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel


class DosenResponse(BaseModel):
    id: int
    user_id: Optional[int]
    nidn: str
    nama: str
    email: str
    prodi: str
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class CreateDosenRequest(BaseModel):
    user_id: Optional[int] = None
    nidn: str
    nama: str
    email: str
    prodi: str


class UpdateDosenRequest(BaseModel):
    nidn: Optional[str] = None
    nama: Optional[str] = None
    email: Optional[str] = None
    prodi: Optional[str] = None
